inc/Engine/Optimization/DeferJS/DeferJS.php

```diff
@@ -289,4 +274,36 @@ public function exclude_jquery_upgrade() {
 
 		update_option( 'wp_rocket_settings', $options );
 	}
+
+	/**
+	 * Get exclusion list pattern.
+	 *
+	 * @return string
+	 */
+	private function get_inline_exclusions_list_pattern() {
+		$inline_exclusions_list = $this->inline_exclusions;
+
+		/**
+		 * Filters the patterns used to find inline JS that should not be deferred
+		 *
+		 * @since 3.8
+		 *
+		 * @param array $inline_exclusions_list Array of inline JS that should not be deferred.
+		 */
+		$additional_inline_exclusions_list = apply_filters( 'rocket_defer_inline_exclusions', null );
+
+		$inline_exclusions = '';
+
+		if ( is_string( $additional_inline_exclusions_list ) ) {
```

Comment:
Wouldn't be better to guard agaist `! is_array()`? 
And always `$additional_inline_exclusions_list` at this point to be converted to array?
Will reduce the numerous type castings from below.

Reply:
this will be good with all types except for objects so when casting object to string and this object didn't implement the magic method `__toString` it will give an error but if u tried to cast this object to array it will work.